src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/DelegateTests.cs

```diff
@@ -0,0 +1,214 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices.JavaScript;
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Runtime.InteropServices.JavaScript.Tests
+{
+    public static class DelegateTests
+    {
+        private static Function _objectPrototype;
+        public static IEnumerable<object[]> Object_Prototype()
+        {
+            _objectPrototype ??= new Function("return Object.prototype.toString;");
+            yield return new object[] { _objectPrototype.Call() };
+        }
+
+        [Fact]
+        public static void MarshalFunction()
+        {
+            HelperMarshal._functionResultValue = 0;
+            HelperMarshal._i32Value = 0;
+
+            Runtime.InvokeJS(@"
+                var funcDelegate = App.call_test_method (""CreateFunctionDelegate"", [  ]);
+                var res = funcDelegate (10, 20);
+                App.call_test_method (""InvokeI32"", [ res, res ]);
+            ");
+
+            Assert.Equal(30, HelperMarshal._functionResultValue);
+            Assert.Equal(60, HelperMarshal._i32Value);
+        }
+
+        [Fact]
+        public static void MarshalFunctionLooptyLoop()
+        {
+            HelperMarshal._functionResultValue = 0;
+            HelperMarshal._i32Value = 0;
+
+            Runtime.InvokeJS(@"
+                var funcDelegate = App.call_test_method (""CreateFunctionDelegate"", [  ]);
+                var res = funcDelegate (10, 20);
+                for (x = 0; x < 1000; x++)
+                {
+                    res = funcDelegate (10, 20);
+                }
+                App.call_test_method (""InvokeI32"", [ res, res ]);
+            ");
+
+            Assert.Equal(30, HelperMarshal._functionResultValue);
+            Assert.Equal(60, HelperMarshal._i32Value);
+        }
+
+        [Fact]
+        public static void MarshalFunctionLooptyLoopIncrement()
```

Comment:
What is it that we want to test here?

Reply:
Calling the delegate in a loop with different values.

Reply:
Marking this resolved as well.  Thought it was a valid test just to cover different tests.